components/Paginated/PaginatedSuppliers.tsx

```diff
@@ -0,0 +1,75 @@
+import { FC } from 'react'
+
+import { Provider } from '@/services/GraphQL/providers/types'
+
+import styles from './Paginated.module.scss'
+
+interface PaginatedSuppliersProps {
+  providers: Provider[]
+  ProvidersPerPage: number
+  totalProvidersLength: number
+  setCurrentPage: (value: number) => void
+  CurrentPage: number
+  indexOfFirstProvider: number
+  indexOfLastProvider: number
+}
+const PaginatedSuppliers: FC<PaginatedSuppliersProps> = ({
+  ProvidersPerPage,
+  totalProvidersLength,
+  setCurrentPage,
+  CurrentPage,
+  providers,
+  indexOfFirstProvider,
+  indexOfLastProvider,
+}) => {
+  const pageNumbers = []
+  for (let i = 1; i <= Math.ceil(totalProvidersLength / ProvidersPerPage); i++) {
+    pageNumbers.push(i)
+  }
+  const slicedProviders = providers.slice(indexOfFirstProvider, indexOfLastProvider)
+  return (
+    <div className={styles.container}>
+      <div className={styles['paginated-container']}>
+        <div className={styles['pages-number']}>
+          <label>
+            {slicedProviders.length * CurrentPage} of {totalProvidersLength}
+          </label>
+        </div>
+        <ul>
+          <div
+            className={styles.button}
+            onClick={() => {
+              if (CurrentPage !== 1) setCurrentPage(CurrentPage - 1)
+            }}
+          >
+            <span>Antes</span>
+          </div>
+          <div className={styles['pages-list']}>
+            {pageNumbers &&
+              pageNumbers.map(page =>
+                CurrentPage === page ? (
+                  <li className={styles.active} key={page} onClick={() => setCurrentPage(page)}>
+                    {page}
+                  </li>
+                ) : (
+                  <li key={page} onClick={() => setCurrentPage(page)}>
+                    {page}
+                  </li>
+                ),
+              )}
+          </div>
+          <div
+            className={styles.button}
+            onClick={() => {
+              if (CurrentPage !== Math.ceil(totalProvidersLength / ProvidersPerPage)) setCurrentPage(CurrentPage + 1)
```

Comment:
What if you extract `Math.ceil(totalProvidersLength / ProvidersPerPage)` into a const?

Reply:
You can use it to create pageNumbers and here

Reply:
the #116 have this too